src/main/CommandHandler.cpp

```diff
@@ -985,24 +986,25 @@ CommandHandler::tx(std::string const& params, std::string& retStr)
         if (transaction)
         {
             // Add it to our current set and make sure it is valid.
-            TransactionQueue::AddResult status =
+            auto [status, payload] =
                 mApp.getHerder().recvTransaction(transaction, true);
 
             root["status"] = TX_STATUS_STRING[static_cast<int>(status)];
             if (status == TransactionQueue::AddResult::ADD_STATUS_ERROR)
             {
                 std::string resultBase64;
-                auto resultBin = xdr::xdr_to_opaque(transaction->getResult());
+                releaseAssertOrThrow(payload);
```

Comment:
The assertion is still there.

Reply:
Sorry I missed this. That being said, I think I agree with Marta re crashing vs. encoding null payload. Since this is a test function only used in loadgen, I think this assert is helpful as it would give us insight into a bad code path. We shouldn't catch it though, so I'll change this to just a `releaseAssert`.